aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -0,0 +1,183 @@
+/// This module supports functionality related to code management.
+module aptos_framework::code {
+    use std::string::String;
+    use std::error;
+    use std::signer;
+    use std::vector;
+
+    // ----------------------------------------------------------------------
+    // Code Publishing
+
+    /// The package registry at the given address.
+    struct PackageRegistry has key {
+        /// Packages installed at this address.
+        packages: vector<PackageMetadata>,
+    }
+
+    /// Metadata for a package.
+    struct PackageMetadata has store, copy, drop {
+        /// Name of this package.
+        name: String,
+        /// The upgrade policy of this package.
+        upgrade_policy: UpgradePolicy,
+        /// The list of modules installed by this package.
+        modules: vector<ModuleMetadata>,
+        /// Address aliases which where used when the modules above were compiled.
+        address_aliases: vector<AddressAlias>,
+        /// Dependencies which were used when this package was compiled.
+        deps: vector<PackageDep>,
```

Comment:
Same question here why do we need this stored on chain if it's already part of the bytecode?

Reply:
No the package deps are not part of the bytecode, only module deps. I wished we would have defined packages as units of deployments in Move from the beginning, but we haven't, so we have those two systems of modules and packages living SxS.